ERCS/erc-7909.md

```diff
@@ -0,0 +1,511 @@
+---
+eip: 7909
+title: Margin Token Wrapper
+description: A general wrapper to convert any token to a lockable token.
+author: Han Chen (@hellohanchen)
+discussions-to: https://ethereum-magicians.org/t/new-erc-margin-token/23222
+status: Draft
+type: Standards Track
+category: ERC
+created: 2025-03-16
+requires: 20, 165, 721, 1155
+---
+
+## Abstract
+
+This proposal defines a token wrapper contract that can convert any [ERC-20](./eip-20.md), [ERC-721](./eip-721.md), [ERC-1155](./eip-1155.md) token to corresponding "margin token", which is an [ERC-1155](./eip-1155.md) token and is lockable.
+
+## Motivation
+
+"Margin" is a widely used concept in trading, it is the collateral that an investor has to deposit with their broker or exchange to cover the credit risk the holder poses for the broker or the exchange. In centralized crypto exchanges, the exchanges will just lock part of users' balances to use as margin. But on blockchain, with decentralized exchanges (DEX), users always need to transfer tokens to a smart contract, then use part of those tokens as margin. There is little interoperability across different DEX protocols and the assets transferred out will not show up on the user's address balance. 
+
+There are existing ERC proposals that suggest a lockable version of tokens. However, those might not be applicable to existing tokens and also each of them is still individual token contracts, the interoperability is still limited. In this proposal, we are defining a wrapper contract that can wrap any token into a "margin token" and can be used across multiple DEX as collaterals. 
```

Comment:
It sounds like a single ERC-1155 contract could be deployed and be used for all lockable tokens on that network. If this is true (and it might not be, I haven't read the entire proposal yet) you shouldn't need to create an ERC. Instead just deploy the contract and write some documentation for its use.

ERCs really shine for ideas that require multiple parties to coordinate without communicating. For example, consider wallets and ERC-20 tokens. Every wallet wants to support every ERC-20 token, while every ERC-20 token wants to be supported by every wallet. Imagine the headache if each pair needed custom code.

In this proposal's case, there could be a single `MarginTokenWrapper` contract on chain, so you don't really have the same M*N explosion you get with tokens/wallets.